In [1]:
# Das Paket tidyverse wird geladen
library(tidyverse)
# Lade den Data-Frame
load(file = "Unterschiedshypothesen1.RData")

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.3     ✔ purrr   0.3.4
✔ tibble  3.1.1     ✔ dplyr   1.0.6
✔ tidyr   1.1.3     ✔ stringr 1.4.0
✔ readr   1.4.0     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()



## Lösung Aufgabe 1.
Damit wir von einer Vergleichbarkeit der beiden Interventionsgruppen ausgehen können, möchten wir überprüfen, ob sich die verschachtelte und die geblockte Bedingung (gruppe) in ihrem Vorwissen (vorwissen) unterscheiden. Überprüft dies.

**Statistisches Verfahren:t-Test für unabhängige Stichproben bzw. Welch-Test für unabhängige Stichproben.**


In [2]:
psych::describeBy(x = Unterschiedshypothesen1$vorwissen, 
                  group = Unterschiedshypothesen1$gruppe)


 Descriptive statistics by group 
group: 0
   vars  n  mean   sd median trimmed  mad min max range  skew kurtosis   se
X1    1 98 13.82 5.78   14.5   14.04 6.67   0  25    25 -0.28    -0.63 0.58
------------------------------------------------------------ 
group: 1
   vars  n  mean   sd median trimmed  mad min max range skew kurtosis  se
X1    1 95 14.12 5.87     14   14.06 7.41   0  26    26 0.04    -0.79 0.6

**Empirische Kennwerte:**

verschachtelt: *M* = 13.82, *SD* = 5.78

geblockt: *M* = 14.12, *SD* = 5.87

In [3]:
# Der Befehl t.test unterstützt unterschiedliche t-test Verfahren.
# Hier wird ein Welch t-test für unabhängige Stichproben durchgeführt.
# Es ist sinvoll die Variable gruppe des Datenframes Unterschiedshypothesen1 als Faktor zu betrachten.
# Dies geschiet über den Befehl as.factor()

t.test(Unterschiedshypothesen1$vorwissen ~ as.factor(Unterschiedshypothesen1$gruppe))

# Das Paket effsize kann mit dem Befehl cohen.d() die Effektstärle Cohen's d berechnen.
# Dabei können alle Parameter des übrigens T-Tests übernommen werden.
# Auch hier ist zu beachten das die Variable gruppe des Datenframes Unterschiedshypothesen1 als Faktor gelesen werden muss

effsize::cohen.d(Unterschiedshypothesen1$vorwissen ~ as.factor(Unterschiedshypothesen1$gruppe))


	Welch Two Sample t-test

data:  Unterschiedshypothesen1$vorwissen by as.factor(Unterschiedshypothesen1$gruppe)
t = -0.35709, df = 190.57, p-value = 0.7214
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.953630  1.354704
sample estimates:
mean in group 0 mean in group 1 
       13.81633        14.11579 



Cohen's d

d estimate: -0.05142713 (negligible)
95 percent confidence interval:
     lower      upper 
-0.3354702  0.2326160 


**Ergebnis:**

t(191) = -0.36, p = .72, d = -0.05

**Interpretation:** Es gibt keinen signifikanten Unterschied im Vorwissen zwischen der
verschachtelten und der geblockten Bedingung – damit sind die beiden Interventionsgruppen
bezüglich ihres Vorwissens vergleichbar. 

## Lösung Aufgabe 2.

Überprüft die Hypothese, dass die Schüler*innen nach der Intervention mehr unterschiedliche
Subtraktionsstrategien verwenden (flex_post) also vor der Intervention (flex_prä).

**Statistisches Verfahren: t-Test für abhängige Stichproben**

In [4]:
psych::describe(Unterschiedshypothesen1 %>% select(flex_prä,flex_post))

vars n   mean      sd        median trimmed   mad    min max range
flex_prä  1    195 0.9435897 0.5849464 1      0.9171975 0.0000 0   3   3    
flex_post 2    195 1.7794872 1.1340966 2      1.8025478 1.4826 0   4   4    
          skew        kurtosis  se        
flex_prä   0.31434533  1.051705 0.04188888
flex_post -0.04866951 -1.085641 0.08121434

**Empirische Kennwerte: Prätest:** 

Prä: *M* = 0.94, *SD* = 0.58

Post: *M* = 1.78, *SD* = 1.13

In [5]:
# Es ist wichtig dem Befehl t.test den Parameter (paired = FALSE) anzufügen, 
#da so ein Welch t-test für verbundene Stichproben berechnent wurd 

t.test(Unterschiedshypothesen1$flex_prä,
       Unterschiedshypothesen1$flex_post,
       paired = TRUE)


effsize::cohen.d(Unterschiedshypothesen1$flex_prä,
                 Unterschiedshypothesen1$flex_post,
                 paired = TRUE)


	Paired t-test

data:  Unterschiedshypothesen1$flex_prä and Unterschiedshypothesen1$flex_post
t = -9.4063, df = 194, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.0111646 -0.6606302
sample estimates:
mean of the differences 
             -0.8358974 



Cohen's d

d estimate: -0.920331 (large)
95 percent confidence interval:
     lower      upper 
-1.1498455 -0.6908164 


**Ergebnis:**
t(194) = -9.41, *p* < .001, *d* = -0.92

**Interpretation:** 
Die Schüler*innen nutzen zum Posttest signifikant mehr unterschiedliche
Strategien als zum Prätest. Dieser Unterschied ist praktisch bedeutsam. Es handelt sich hierbei
mit Cohen’s *d* = -0.92 um einen großen Effekt.